pycbc/io/hdf.py

```diff
@@ -713,14 +713,19 @@ def trig_id(self):
         if self._trig_ids is not None:
             return self._trig_ids
         self._trig_ids = {}
-        # FIXME: There is no clear mapping from trig_id to ifo. This is bad!!!
-        #        for now a hack is in place.
-        ifo1 = self.coinc_file.h5file.attrs['detector_1']
-        ifo2 = self.coinc_file.h5file.attrs['detector_2']
-        trigid1 = self.get_coincfile_array('trigger_id1')
-        trigid2 = self.get_coincfile_array('trigger_id2')
-        self._trig_ids[ifo1] = trigid1
-        self._trig_ids[ifo2] = trigid2
+
+        try: # New style multi-ifo file
+            ifos = self.coinc_file.h5file.attrs['ifos'].split(' ')
+            for ifo in ifos:
+                trigid = self.get_coincfile_array(ifo + '/trigger_id')
+                self._trig_ids[ifo] = trigid
+        except: # ADD ERROR # Old style two-ifo file
+            ifo1 = self.coinc_file.h5file.attrs['detector_1']
+            ifo2 = self.coinc_file.h5file.attrs['detector_2']
+            trigid1 = self.get_coincfile_array('trigger_id1')
+            trigid2 = self.get_coincfile_array('trigger_id2')
+            self._trig_ids[ifo1] = trigid1
+            self._trig_ids[ifo2] = trigid2
```

Comment:
I think that these things before have been set by something like 
```
if 'ifos' in self.coinc_file.h5file.attrs:
    ifos=self.coinc_file.h5file.attrs['ifos'].split(' ')
else:
    ifos=[self.coinc_file.h5file.attrs['detector_1'], self.coinc_file.h5file.attrs['detector_2']]
```
 and then everything else can remain the same as the multiifo case

Reply:
No, that's not true because the `trigger_id` is stored in different places. My logic here is to leave the old code alone as much as possible, and then we can just delete it at some point.